federation/executor.go

```diff
@@ -59,72 +95,52 @@ func fetchSchema(ctx context.Context, e ExecutorClient, metadata interface{}) (*
 	})
 }
 
-func NewExecutor(ctx context.Context, executors map[string]ExecutorClient) (*Executor, error) {
-	return NewExecutorWithOptionalArgs(ctx, executors, nil)
+type CustomExecutorArgs struct {
+	schemaSyncer             SchemaSyncer
+	optionalArgs             interface{}
+	schemaSyncIntervalSecond int64
```

Comment:
Should we expose these fields? Since we need to use them outside of thunder. 
Also, do you think we can move the `schemaSyncIntervalSecond` and the poller function to the syncer?

Reply:
The executor is the object that has the ticker and is polling the schema syncer so it would be misleading to put it into the schema syncer. The schema syncer is also just an interface with a method to fetch the planner - so I would prefer to leave schemaSyncIntervalSecond in the executor.

Reply:
Then how about change `schemaSyncIntervalSecond` to a function that returns an int64 which gives the user more flexibility to control the interval realtime?

Reply:
Ah thats a really good point! I added it as a function to schema syncer!